api/autoscale.go

```diff
@@ -0,0 +1,80 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/labstack/echo/v4"
+	"github.com/tsuru/rpaas-operator/internal/pkg/rpaas"
+)
+
+func getAutoscale(c echo.Context) error {
+	manager, err := getManager(c)
+	if err != nil {
+		return err
+	}
+
+	autoscale, err := manager.GetAutoscale(c.Request().Context(), c.Param("instance"))
+	if err != nil {
+		return err
+	}
+
+	if autoscale == nil {
+		autoscale = &rpaas.Autoscale{}
+	}
+
+	return c.JSON(http.StatusOK, struct {
+		Autoscale *rpaas.Autoscale `json:"autoscale"`
+	}{autoscale})
+}
+
+func createAutoscale(c echo.Context) error {
+	manager, err := getManager(c)
+	if err != nil {
+		return err
+	}
+
+	var autoscale rpaas.Autoscale
+	if err = c.Bind(&autoscale); err != nil {
+		return err
+	}
+
+	err = manager.CreateAutoscale(c.Request().Context(), c.Param("instance"), &autoscale)
+	if err != nil {
+		return err
+	}
+
+	return c.NoContent(http.StatusOK)
+}
+
+func updateAutoscale(c echo.Context) error {
+	manager, err := getManager(c)
+	if err != nil {
+		return err
+	}
+
+	var autoscale rpaas.Autoscale
```

Comment:
To be able to handle correctly the PATCH verb, we need to merge the current state of Autoscale with the modified by the user in this request before to call the `UpdateAustocale` method. Without this change, the user can reach an undesired autoscale configuration since the unchanged arguments will be set as nil into `UpdateAutoscale`.

Another option would be turning this handler a PUT handler and require the user to send the whole autoscale object.

Reply:
Ops, totally overlooked this while implementing, i'll fix it this.

Reply:
fixed!